### ISSUE TITLE
How does play end?

### ISSUE THREAD

**Issue by jlmitch1989** [ga-wdi-boston/game-project#233]
Here's my click handler:

``` js
$('.game-tile').on('click', onTurn);
```

Here's my boardFull function so far:

`````` js
const boardFull = function () {
  for (let i = 0; i < 9; i++) {
    return gameBoardArray[i];
  }


Here's my getWinner function:

```js
const getWinner = function() {
if (gameBoardArray[0] === 'x' &&      //Start of rows for 'x'
    gameBoardArray[1] === 'x' &&
    gameBoardArray[2] === 'x') {
      console.log('X WINS');
      return 'x';
    }
if (gameBoardArray[3] === 'x' &&
    gameBoardArray[4] === 'x' &&
    gameBoardArray[5] === 'x') {
      return 'x';
    }
if (gameBoardArray[6] === 'x' &&
    gameBoardArray[7] === 'x' &&
    gameBoardArray[8] === 'x') {
      return 'x';
    }
if (gameBoardArray[0] === 'x' &&      //Start of columns for 'x'
    gameBoardArray[3] === 'x' &&
    gameBoardArray[6] === 'x') {
      return 'x';
    }
if (gameBoardArray[1] === 'x' &&
    gameBoardArray[4] === 'x' &&
    gameBoardArray[7] === 'x') {
      return 'x';
    }
if (gameBoardArray[2] === 'x' &&
    gameBoardArray[5] === 'x' &&
    gameBoardArray[8] === 'x') {
      return 'x';
    }
if (gameBoardArray[0] === 'x' &&        //Start of diagonals for 'x'
    gameBoardArray[4] === 'x' &&
    gameBoardArray[8] === 'x') {
      return 'x';
    }
if (gameBoardArray[2] === 'x' &&
    gameBoardArray[4] === 'x' &&
    gameBoardArray[6] === 'x') {
      return 'x';
    }
if (gameBoardArray[0] === 'o' &&      //Start of rows for 'o'
    gameBoardArray[1] === 'o' &&
    gameBoardArray[2] === 'o') {
      return 'o';
    }
if (gameBoardArray[3] === 'o' &&
    gameBoardArray[4] === 'o' &&
    gameBoardArray[5] === 'o') {
      return 'o';
    }
if (gameBoardArray[6] === 'o' &&
    gameBoardArray[7] === 'o' &&
    gameBoardArray[8] === 'o') {
      return 'o';
    }
if (gameBoardArray[0] === 'o' &&      //Start of columns for 'o'
    gameBoardArray[3] === 'o' &&
    gameBoardArray[6] === 'o') {
      return 'o';
    }
if (gameBoardArray[1] === 'o' &&
    gameBoardArray[4] === 'o' &&
    gameBoardArray[7] === 'o') {
      return 'o';
    }
if (gameBoardArray[2] === 'o' &&
    gameBoardArray[5] === 'o' &&
    gameBoardArray[8] === 'o') {
      return 'o';
    }
if (gameBoardArray[0] === 'o' &&        //Start of diagonals for 'o'
    gameBoardArray[4] === 'o' &&
    gameBoardArray[8] === 'o') {
      return 'o';
    }
if (gameBoardArray[2] === 'o' &&
    gameBoardArray[4] === 'o' &&
    gameBoardArray[6] === 'o') {
      return 'o';
    }
  };
``````

Here's my

**Comment by jlmitch1989** [comment]
Sorry, here's my getWinner function again:

``` js
const getWinner = function() {
if (gameBoardArray[0] === 'x' &&      //Start of rows for 'x'
    gameBoardArray[1] === 'x' &&
    gameBoardArray[2] === 'x') {
      console.log('X WINS');
      return 'x';
    }
if (gameBoardArray[3] === 'x' &&
    gameBoardArray[4] === 'x' &&
    gameBoardArray[5] === 'x') {
      return 'x';
    }
if (gameBoardArray[6] === 'x' &&
    gameBoardArray[7] === 'x' &&
    gameBoardArray[8] === 'x') {
      return 'x';
    }
if (gameBoardArray[0] === 'x' &&      //Start of columns for 'x'
    gameBoardArray[3] === 'x' &&
    gameBoardArray[6] === 'x') {
      return 'x';
    }
if (gameBoardArray[1] === 'x' &&
    gameBoardArray[4] === 'x' &&
    gameBoardArray[7] === 'x') {
      return 'x';
    }
if (gameBoardArray[2] === 'x' &&
    gameBoardArray[5] === 'x' &&
    gameBoardArray[8] === 'x') {
      return 'x';
    }
if (gameBoardArray[0] === 'x' &&        //Start of diagonals for 'x'
    gameBoardArray[4] === 'x' &&
    gameBoardArray[8] === 'x') {
      return 'x';
    }
if (gameBoardArray[2] === 'x' &&
    gameBoardArray[4] === 'x' &&
    gameBoardArray[6] === 'x') {
      return 'x';
    }
if (gameBoardArray[0] === 'o' &&      //Start of rows for 'o'
    gameBoardArray[1] === 'o' &&
    gameBoardArray[2] === 'o') {
      return 'o';
    }
if (gameBoardArray[3] === 'o' &&
    gameBoardArray[4] === 'o' &&
    gameBoardArray[5] === 'o') {
      return 'o';
    }
if (gameBoardArray[6] === 'o' &&
    gameBoardArray[7] === 'o' &&
    gameBoardArray[8] === 'o') {
      return 'o';
    }
if (gameBoardArray[0] === 'o' &&      //Start of columns for 'o'
    gameBoardArray[3] === 'o' &&
    gameBoardArray[6] === 'o') {
      return 'o';
    }
if (gameBoardArray[1] === 'o' &&
    gameBoardArray[4] === 'o' &&
    gameBoardArray[7] === 'o') {
      return 'o';
    }
if (gameBoardArray[2] === 'o' &&
    gameBoardArray[5] === 'o' &&
    gameBoardArray[8] === 'o') {
      return 'o';
    }
if (gameBoardArray[0] === 'o' &&        //Start of diagonals for 'o'
    gameBoardArray[4] === 'o' &&
    gameBoardArray[8] === 'o') {
      return 'o';
    }
if (gameBoardArray[2] === 'o' &&
    gameBoardArray[4] === 'o' &&
    gameBoardArray[6] === 'o') {
      return 'o';
    }
  };
```

**Comment by laurenfazah** [comment]
So question is how does game play end, right? Looks like this function returns values `x` or `o` if one of those wins, so that's one way you can end (if that comes back with a value). Another way is if there's a draw. Do you have a way of counting the turns made? What could you leverage that for to see if all possible moves have been made without a winner?

**Comment by jlmitch1989** [comment]
Can you rephrase the question?  I can't seem to understand what your asking.

**Comment by laurenfazah** [comment]
No problem. 

You're asking how to end game play, right?

Well typically, it's either by someone winning, or the game being a draw. Looks like you've got a check for winning - is that working?

If so, are you able to check if the game is a draw (aka all moves are made and no one has won)?

**Comment by jlmitch1989** [comment]
I tried testing to see if the code for 'x' or 'o' wins was working but I couldn't get anything to print to my console. I don't know if I was doing it right or not.

And for the last question, that's mostly what I'm having issues with; trying to figure out a function that can tell me that.

**Comment by laurenfazah** [comment]
To get things to print to the console, you'd want to `console.log()`, not `return`. What is invoking your `getWinner()` function?

**Comment by jlmitch1989** [comment]
It thought that's where I was going with my boardFull function.

**Comment by laurenfazah** [comment]
Doesn't look like `boardFull` is calling `getWinner`. What does your `onTurn` function look like? The one being called by the event listener?

**Comment by jlmitch1989** [comment]
I did use console.log to test to see if the 1st row for 'x' was working but it didn't work. 

I'm not sure I have anything invoking the getWinner function yet.

**Comment by jlmitch1989** [comment]
``` js
const onTurn = function () {
  let $tile = $(this);
  if ($tile.attr('style')) {
    return;
  }else {
    if (counter % 2) {
      //first person logic
      $tile.css("background-image" , 'url("' + ticTacPic + '")');
      let index = parseInt($tile.data('index'), 10);
      gameBoardArray[index] = 'x';
      // console.log('first person');
    }else {
      //second person logic
      $tile.css("background-image" , 'url("' + toePic + '")');
      let index = parseInt($tile.data('index'), 10);
      gameBoardArray[index] = 'o';
      // console.log('second person');
    }
      counter++;
      // console.log('The count is ' + counter);
    if (counter < 3) {
      return;
    }
    //Check for winner function

    }
};
```

**Comment by laurenfazah** [comment]
See your comment for `//Check for winner function` there? Looks like a good place to call the `getWinner` function :)

**Comment by jlmitch1989** [comment]
Ok well that worked, but I still don't understand how to write out this function for a draw

**Comment by laurenfazah** [comment]
In your `onTurn` function, you're increasing the `counter` value each time a move is made, correct? Do you think you can leverage this to assume a draw if a certain amount of moves have been made, but a winner has not been found?

**Comment by jlmitch1989** [comment]
Yes, by a for loop, which I have attempted to start here(w/ the help of Antony):

``` js
const boardFull = function () {
  for (let i = 0; i < 9; i++) {
    return gameBoardArray[i];
  }
```

I just don't fully understand where to go with it.

**Comment by laurenfazah** [comment]
Hmm, doesn't look like that's increasing `counter`'s value. You're doing it right in `onTurn` with `counter++;`

**Comment by laurenfazah** [comment]
You could simply check `counter`'s value at any point to see if it's reached its max. If it has, and there's no winner, looks like you've got a draw.

**Comment by jlmitch1989** [comment]
``` js
const boardFull = function () {
  for (let i = 0; i < gameBoardArray[].length; i++) {
    return gameBoardArray[i];
  }
``
like this?
```

**Comment by laurenfazah** [comment]
Is that doing anything with `counter`?

**Comment by laurenfazah** [comment]
I'm not sure where you and @gaand were going with `boardFull`, but it doesn't appear to be doing more than just looping (before over 9 numbers, now over the `gameBoardArray`), so I wouldn't focus too heavily on it right now.

**Comment by jlmitch1989** [comment]
Yeah I thought I knew where we were going but apparently I was mistaken :(

**Comment by gaand** [comment]
You want to check for a winner, and if there isn't one, check for board full.  I'd put those checks in an if/else near the end of your `onTurn` handler to start.

**Comment by laurenfazah** [comment]
Want to come to the dugout @jlmitch1989 ?

**Comment by jlmitch1989** [comment]
One moment. I'm trying something.

**Comment by laurenfazah** [comment]
Josh and I went through and psuedo-coded the remainder of what is needed for to complete game play. He's on track to wrap this up himself.
